### PR TITLE
MBS-10365 (IV): Convert entity lists to react-table

### PR DESCRIPTION
**Pull request by reosarevok**
MBS-10365

* [ ] Artist
* [x] Event
* [ ] Recording
* [x] Release
* [x] Release group

### CODE REVIEW

File: root/utility/tableColumns.js [hunk 1]
```diff
@@ -354,7 +355,7 @@ export function defineSeriesNumberColumn(
 }
 
 export function defineTextColumn<D>(
-  getText: (D) => string,
+  getText: (D) => StrOrNum,
```
Comment (yvanzo) [inline root/utility/tableColumns.js]: Related to the previous comments, it seems safer to just allow `string` to enforce formatting numbers for display.

File: root/components/list/ReleaseGroupList.js [hunk 2]
```diff
@@ -61,155 +44,90 @@ type ReleaseGroupListProps = {
...
+        ? defineSeriesNumberColumn(seriesItemNumbers)
+        : null;
+      const yearColumn = defineTextColumn(
+        entity => parseDate(entity.firstReleaseDate).year ?? '—',
+        'release_count',
+        l('Year'),
+        order,
+        sortable,
+        'year c',
+      );
```
[the remark this answers is not in the document]
Comment (reosarevok) [reply root/components/list/ReleaseGroupList.js]: Do we actually have any code to format years to non-Latin locales?

[the remark this answers is not in the document]
Comment (yvanzo) [reply root/components/list/ReleaseGroupList.js]: Theoretically the same example as above: https://github.com/metabrainz/musicbrainz-server/blob/v-2020-03-16/root/statistics/utilities.js#L26-L30 but I did not try to make it work for non-Latin locales yet, it probably requires to install additional language packs.

[the remark this answers is not in the document]
Comment (yvanzo) [reply root/components/list/ReleaseGroupList.js]: Found out that locales are fully supported in Node.js v13 only.
Previous versions should either be compiled with custom options or use `full-icu` module.

[the remark this answers is not in the document]
Comment (yvanzo) [reply root/components/list/ReleaseGroupList.js]: That might explain why `formatCount` is currently not working…

**Pull request merged**